SU2_CFD/include/variables/CVariable.hpp

```diff
@@ -2112,80 +2127,43 @@ class CVariable {
   /*!
    * \brief Register the variables in the solution array as input/output variable.
    * \param[in] input - input or output variables.
-   * \param[in] push_index - boolean whether we want to push the index or save it in a member variable.
    */
-  void RegisterSolution(bool input, bool push_index = true);
```

Comment:
It is still used by a few variables.

Reply:
ah yes, by everything in `CDiscAdjSolver::RegisterVariables` i.e. Mach, Freestream values etc. As far as I see all those vars are unaffected by the fixed-point update so their original value/variable stays in memory -> so it should be possible to go without that logic . That would of course mean the `GetDerivative(const su2double& data)` has to be changed from 
```c++
  FORCEINLINE passivedouble GetDerivative(const su2double& data) {
    return AD::getGlobalTape().getGradient(AD::inputValues[AD::adjointVectorPosition++]);
  }
```
to the one below
```c++
FORCEINLINE passivedouble GetDerivative(const su2double& data) {return data.getGradient();}
```
RN there is also this fake input `const su2double& data` where one thinks one could directly extract fom a variable like so
```c++
Local_Sens_Mach  = SU2_TYPE::GetDerivative(Mach);
```
which gives a false sense of whats happening because one would expect it works like the proposed one (I certainly did). It should also made clear in those two functions (register & extract) that the order of registering has to match the order of extracting. that is currently not the case.

In case the `inputValues` is kept I would like to see the `GetDerivative(const su2double& data) ` be removed, added explanation in Register/Extract-Variables and maybe switch the default `push_index=True` to `False` in `AD::RegisterInput` to have this additional way to say "look-what-that-boolean-does" and then everyone should be aware that extract the same way you register is what should be done

(When I say `one should do sth` in the text or anywhere I am happy to do it myself but I'll ask first if I overlook sth in my ideas)

Reply:
Ah yes, I remember the day I saw that for the first time... Sadly I cannot reproduce the words I hurled at the screen here xD
Anyway, let's see if it works...